### PR TITLE
New-DbaCertificate

### PR DESCRIPTION
**Pull request by potatoqualitee**
Looks good eh! Returns a Certificate object. 

image

### CODE REVIEW

File: functions/New-DbaComputerCertificate.ps1 [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+		if (Test-FunctionInterrupt) { return }
+		
+		foreach ($computer in $computername) {
+			if (!$secondarynode) {
+				if (![dbavalidate]::IsLocalhost($computer) -and !$Password) {
+					Write-Message -Level Output -Message "You have specified a remote computer. A password is required for private key encryption/decryption for import."
+					$Password = Read-Host -AsSecureString -Prompt "Password"
+				}
+				
+				if ($InstanceClusterName) {
```
Comment (PowerDBAKlaas) [inline functions/New-DbaComputerCertificate.ps1]: InstanceClusterName
aren't these Windows clusters?

File: functions/New-DbaComputerCertificate.ps1 [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+.NOTES
+Tags: Certificate
+
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.EXAMPLE
+New-DbaCertificate
```
Comment (PowerDBAKlaas) [inline functions/New-DbaComputerCertificate.ps1]: New-DbaCertificate -> New-DbaComputerCertificate

File: functions/New-DbaComputerCertificate.ps1 [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+		[Alias("ServerInstance", "SqlServer", "SqlInstance")]
+		[DbaInstanceParameter[]]$ComputerName = $env:COMPUTERNAME,
+		[System.Management.Automation.PSCredential]$Credential,
+		[string]$CaServer,
+		[string]$CaName,
+		[string]$InstanceClusterName,
+		[securestring]$Password,
+		[string]$FriendlyName = "SQL Server",
+		[string]$CertificateTemplate = "WebServer",
+		[int]$KeyLength = 1024,
```
Comment (potatoqualitee) [inline functions/New-DbaComputerCertificate.ps1]: It'd actually be best doing divisible by 1024 but I'm too exhausted. Right now it'll just error out.

File: functions/New-DbaComputerCertificate.ps1 [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+		[switch]$Silent
+	)
+	begin {
+		
+		Test-RunAsAdmin
+		
+		if (!$CaServer -or !$CaName) {
```
Comment (potatoqualitee) [inline functions/New-DbaComputerCertificate.ps1]: Wait, why? if either are missing then it needs to populate.

File: functions/New-DbaComputerCertificate.ps1 (hunk 1 above)
Comment (potatoqualitee) [inline functions/New-DbaComputerCertificate.ps1]: Yes, it's a Windows cluster (wincluster in my AD), but it's also a SQL Instance Cluster (sqlcluster in my AD) - the SQL cluster name is what needs to be encrypted.

File: functions/New-DbaComputerCertificate.ps1 (hunk 4 above)
Comment (potatoqualitee) [inline functions/New-DbaComputerCertificate.ps1]: Thanks for the review btw!

**Pull request merged**